packages/auth/amplify_auth_cognito/example/integration_test/confirm_sign_up_test.dart

```diff
@@ -0,0 +1,156 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:amplify_test/amplify_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/mock_data.dart';
+import 'utils/setup_utils.dart';
+import 'utils/test_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group(
+    'confirmSignUp',
+    () {
+      setUpAll(() async {
+        await configureAuth(
+          additionalPlugins: [AmplifyAPI()],
+        );
+      });
+
+      setUp(signOutUser);
+
+      asyncTest('can confirm sign up', (_) async {
+        final username = generateUsername();
+        final password = generatePassword();
+
+        // Sign up, but do not confirm, user
+        final code = getOtpCode(username);
+        final signUpResult = await Amplify.Auth.signUp(
+          username: username,
+          password: password,
+          options: CognitoSignUpOptions(
+            userAttributes: {
+              CognitoUserAttributeKey.email: generateEmail(),
+              CognitoUserAttributeKey.phoneNumber: mockPhoneNumber
+            },
+          ),
+        );
+        expect(signUpResult.isSignUpComplete, false);
+        expect(
+          signUpResult.nextStep.codeDeliveryDetails?.deliveryMedium,
+          'SMS',
+        );
+
+        // Confirm sign up and complete sign in
+        final confirmResult = await Amplify.Auth.confirmSignUp(
+          username: username,
+          confirmationCode: await code,
+        );
+        expect(confirmResult.isSignUpComplete, true);
+      });
+
+      asyncTest('can sign up after sign in', (_) async {
+        final username = generateUsername();
+        final password = generatePassword();
+
+        // Sign up, but do not confirm, user
+        final code = getOtpCode(username);
+        final signUpResult = await Amplify.Auth.signUp(
+          username: username,
+          password: password,
+          options: CognitoSignUpOptions(
+            userAttributes: {
+              CognitoUserAttributeKey.email: generateEmail(),
+              CognitoUserAttributeKey.phoneNumber: mockPhoneNumber
+            },
+          ),
+        );
+        expect(signUpResult.isSignUpComplete, false);
+        expect(
+          signUpResult.nextStep.codeDeliveryDetails?.deliveryMedium,
+          'SMS',
+        );
```

Comment:
We should put redundant code into a separate method to reduce code duplication.  In this case it looks like the sign up code is the same between both methods.